js/app/promiseWrapper.js

```diff
@@ -0,0 +1,44 @@
+const Eknc = imports.gi.EosKnowledgeContent;
+const Gio = imports.gi.Gio;
+const Soup = imports.gi.Soup;
+
+function wrapPromise (obj, cancellable, async_name) {
```

Comment:
Moved away from supporting cancellation directly in promises for now. Reading more (here http://bluebirdjs.com/docs/api/cancellation.html for example) there's some tough cases where you have multiple consumers of the same promise.

Easier to just take in an optional GCancellable when calling into one of these `_promise` functions and handle passing the cancellable manually. In our case that's only with downloader.js